src/context/trace_context.rs

```diff
@@ -14,387 +14,392 @@
+    ) -> T {
+        f(&mut *self.inner.active_span_stack.try_lock().expect(LOCK_MSG))
     }
 
-    pub fn from_propagation_context_internal(
-        time_fetcher: Arc<dyn TimeFetcher + Sync + Send>,
-        service_name: &str,
-        instance_name: &str,
-        context: PropagationContext,
-    ) -> Self {
-        TracingContext {
-            trace_id: context.parent_trace_id.clone(),
-            trace_segment_id: RandomGenerator::generate(),
-            service: service_name.to_string(),
-            service_instance: instance_name.to_string(),
-            next_span_id: 0,
-            time_fetcher,
-            spans: Vec::new(),
-            segment_link: Some(context),
-            active_span_id_stack: LinkedList::new(),
-        }
+    pub(crate) fn try_with_active_span_stack<T>(
+        &self,
+        f: impl FnOnce(&Vec<SpanObject>) -> T,
+    ) -> Option<T> {
+        self.inner
+            .active_span_stack
+            .try_lock()
+            .ok()
+            .map(|stack| f(&*stack))
     }
 
-    /// A wrapper of create entry span, which close generated span automatically.
-    /// Note that, we may use async operation in closure. But that is not unstable feature in 2021/12.
-    /// <https://github.com/rust-lang/rust/issues/62290>
-    /// So we should create and close spans manually in general.
-    pub fn entry<F: FnMut(&Span)>(
-        &mut self,
-        operation_name: &str,
-        mut process_fn: F,
-    ) -> crate::Result<()> {
-        match self.create_entry_span(operation_name) {
-            Ok(mut span) => {
-                process_fn(span.as_ref());
-                span.close();
-                Ok(())
-            }
-            Err(message) => Err(message),
-        }
+    pub(crate) fn with_active_span<T>(&self, f: impl FnOnce(&SpanObject) -> T) -> Option<T> {
+        self.with_active_span_stack(|stack| stack.last().map(f))
+    }
+
+    // TODO Using for capture and continued.
+    #[allow(dead_code)]
+    fn with_primary_endpoint_name<T>(&self, f: impl FnOnce(&String) -> T) -> T {
+        f(&*self.inner.primary_endpoint_name.try_lock().expect(LOCK_MSG))
+    }
+
+    fn with_primary_endpoint_name_mut<T>(&mut self, f: impl FnOnce(&mut String) -> T) -> T {
+        f(&mut *self.inner.primary_endpoint_name.try_lock().expect(LOCK_MSG))
     }
 
     /// Create a new entry span, which is an initiator of collection of spans.
     /// This should be called by invocation of the function which is triggered by
     /// external service.
-    pub fn create_entry_span(&mut self, operation_name: &str) -> crate::Result<Box<Span>> {
-        if self.next_span_id >= 1 {
-            return Err(crate::Error::CreateSpan("entry span have already exist."));
+    ///
+    /// # Panics
+    ///
+    /// Panic if entry span have already exist.
+    pub fn create_entry_span(&mut self, operation_name: &str) -> Span {
+        if self.next_span_id() >= 1 {
+            panic!("entry span have already exist.");
```

Comment:
Use panic instead of returning error, because it is caused by the programmer's incorrect usage, not external input. If there is a need to return an error in the future, can add a `try_` prefix method.